### ISSUE TITLE
RS algorithm encoding part mismatch signature. 

### ISSUE THREAD

**Issue by lolgear** [yourkarma/JWT#114]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/yourkarma/JWT/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](https://github.com/yourkarma/JWT#documentation)
* [x] I have searched for a similar issue in the [project](https://github.com/yourkarma/JWT/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / osx / tvos / watchos
 Platform Version        | e.g. 8.0
 CocoaLumberjack Version | e.g. 2.3.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Curiously:

RS Algorithm generates different signature on existing test data while encoding.
However, decoding part works smoothly.

**Comment by lolgear** [comment]
https://github.com/yourkarma/JWT/commit/7cc43d817522b5e37455ccca35c9c2f9084acfea
